src/include/api/vamana_index.h

```diff
@@ -131,10 +128,19 @@ class IndexVamana {
       const tiledb::Context& ctx,
       const URI& group_uri,
       std::optional<TemporalPolicy> temporal_policy = std::nullopt) {
-    read_types(ctx, group_uri, &feature_datatype_, &id_datatype_);
-
+    tiledb_datatype_t distance_metric_raw;
+    read_types(
+        ctx,
+        group_uri,
+        &feature_datatype_,
+        &id_datatype_,
+        &distance_metric_raw);
+    distance_metric_ = static_cast<DistanceMetric>(distance_metric_raw);
```

Comment:
(Minor) This is okay, but why not just have `read_types()` return `DistanceMetric`? Seems cleaner?

Reply:
we would have to move the distance_metric_raw logic inside the read_types and i thought it would be different than the pattern we already have - i can definitely move it there, it just seemed cleaner

Reply:
Either way, your choice